### PR TITLE
setup env workflow

### PR DESCRIPTION
**Pull request by JustinCast**
### GH Issue

cl0ses #541 

### Steps to test
1. `make run`

### Checklist
- [ ] I have added/updated tests to cover these changes.
- [ ] The branch name, commit history and PR title follow [conventions](https://developers.eoscostarica.io/docs/open-source-guidelines#pull-request-general-guidelines).
- [ ] I have taken into consideration any impact to site performance.

### CODE REVIEW

File: .github/workflows/push-master-environment.yaml [hunk 1]
```diff
@@ -70,8 +70,8 @@ jobs:
           DB_PORT: "5432"
           DB_SCHEMA: ${{ secrets.DB_SCHEMA }}
           EOS_API_ENDPOINT: https://bp.cryptolions.io
-          VIRTUAL_HOST: 0.0.0.0
-          VIRTUAL_PORT: "9090"
+          VIRTUAL_HOST: ${{ secrets.VIRTUAL_HOST }}
+          VIRTUAL_PORT: ${{ secrets.VIRTUAL_PORT }}
```
Comment (kuronosec) [inline .github/workflows/push-master-environment.yaml:74]: We have to make a balance between what data really needs to be private and the transparency for the developer, does the developer always has access to the secrets?

File: services/frontend/makefile [hunk 2]
```diff
@@ -22,6 +22,13 @@ build-docker: ./Dockerfile
 	@docker build \
 		-t $(DOCKER_REGISTRY)/$(IMAGE_NAME_WEBAPP):$(VERSION) --target server \
 		-t $(DOCKER_REGISTRY)/$(IMAGE_NAME_WEBAPP):$(LATEST_TAG) --target server \
+		--build-arg react_app_graphql_http_url="$(REACT_APP_GRAPHQL_HTTP_URL)" \
+		--build-arg react_app_graphql_ws_url="$(REACT_APP_GRAPHQL_WS_URL)" \
+		--build-arg react_app_eos_api_url=$(REACT_APP_EOS_API_URL) \
```
Comment (kuronosec) [inline services/frontend/makefile:27]: We also need the double quotes here.

File: services/frontend/package.json [hunk 3]
```diff
@@ -11,6 +11,7 @@
     "lint": "eslint .",
     "start": "env-cmd -f .env react-scripts start",
     "start:staging": "env-cmd -f .env.staging react-scripts start",
+    "start:local": "env-cmd -f ../../.env react-scripts start",
```
Comment (kuronosec) [inline services/frontend/package.json:14]: Whats the difference between `start` and `start:local`?

File: services/frontend/package.json [hunk 4]
```diff
@@ -11,6 +11,7 @@
...
     "prod": "serve -l 9000 --single ./build",
     "build": "env-cmd -f .env react-scripts build",
     "build:staging": "env-cmd -f .env.staging react-scripts build",
```
Comment (kuronosec) [inline services/frontend/package.json:17]: Wasn't the idea to get rid of .env.* ?

**Review by kuronosec — CHANGES_REQUESTED** [review]

File: .github/workflows/push-master-environment.yaml (hunk 1 above)
Comment (JustinCast) [reply .github/workflows/push-master-environment.yaml:74]: @xavier506

Comment (xavier506) [reply .github/workflows/push-master-environment.yaml:74]: I think it is easier to maintain and safer if  non-sensitive env vars values are assigned  in the yaml file

**Comment by xavier506** [comment]
@JustinCast do you think we should leave all env vars as secrets?

**Review by kuronosec — APPROVED** [review]
LGTM

**Comment by JustinCast** [comment]
@xavier506 Which ones do you think we should keep as fixed values?

**Pull request merged**